src/components/SignUpPopup/SignUpPopup.js

```diff
@@ -0,0 +1,169 @@
+import { useState } from "react";
+import PopupWithForm from "../PopupWithForm/PopupWithForm";
+import "../SignInPopup/SignInPopup.css";
+import "../SignUpPopup/SignUpPopup.css";
+import * as auth from "../../utils/auth";
+import validation from "../../utils/validate";
+
+function SignUpPopup({ isOpen, onClose, onSignUp, openSignInPopup }) {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [name, setName] = useState("");
+  const [submitErrorMsg, setSubmitErrorMsg] = useState("");
+  let activeSubmit=false;
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [nameError, setNameError] = useState("");
+  if(email && password && name && emailError ==="" && passwordError==="" && nameError===""){
+    activeSubmit=true;
+  }
+  
+  
+  function onEmailChange(e) {
+    const currentEmailInput = e.target.value;
+    setEmail(currentEmailInput);
+    if (
+      currentEmailInput.length > 0 &&
+      !validation(currentEmailInput, "email")
+    ) {
+      setEmailError("Provide a valid email address");
+    } else if (
+      (currentEmailInput.length > 0 &&
+        validation(currentEmailInput, "password")) ||
+      currentEmailInput.length === 0
+    ) {
+      setEmailError("");
+    }
+
+  }
+
+  function onPasswordChange(e) {
+    const currentPasswordInput = e.target.value;
+    setPassword(currentPasswordInput);
+    if (
+      currentPasswordInput.length > 0 &&
+      !validation(currentPasswordInput, "password")
+    ) {
+      setPasswordError("Password must consist of at least 8 characters");
+    } else if (
+      (currentPasswordInput.length > 0 &&
+        validation(currentPasswordInput, "password")) ||
+      currentPasswordInput.length === 0
+    ) {
+      setPasswordError("");
+    }
+  }
+
+  function onNameChange(e){
+    const currentNameInput = e.target.value;
+    setName(currentNameInput);
+    if (
+      currentNameInput.length > 0 &&
+      !validation(currentNameInput, "name")
+    ) {
+      setNameError("Name must have at least 2 characters");
+    } else if (
+      (currentNameInput.length > 0 &&
+        validation(currentNameInput, "name")) ||
+      currentNameInput.length === 0
+    ) {
+      setNameError("");
+    }
+  }
+
+  
+
+  function resetForm() {
+    setEmail("");
+    setPassword("");
+    setName("");
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    auth
+      .register(password, email, name)
+      .then(()=>{
+        onClose();
+        onSignUp(true);
+        resetForm();
+      })
+      .catch((err) => {
+        console.log(err);
+        if(err ==='Error: Error 409')
+        {setSubmitErrorMsg('This email is not available')}
+        else {setSubmitErrorMsg('Problem with the server try again')}
+        const inputList = Array.from(document.getElementsByClassName('popup__form-input'));
```

Comment:
`        const inputList = Array.from(document.getElementsByClassName('popup__form-input'));
`


it's a `pure js` variant 


you should **not` search anything in the DOM in React. It was specially created not to do that because it's a hard process for the browser engine